src/app/features/home/capture-tab/capture-item/capture-item.component.ts

```diff
@@ -90,37 +88,30 @@ export class CaptureItemComponent {
     private readonly router: Router,
     private readonly route: ActivatedRoute,
     private readonly sanitizer: DomSanitizer,
-    private readonly diaBackendAssetRepository: DiaBackendAssetRepository,
-    private readonly toastController: ToastController,
-    private readonly translocoService: TranslocoService
+    private readonly diaBackendAssetRepository: DiaBackendAssetRepository
   ) {}
 
   @HostListener('click')
   async onClick() {
-    const hasUploaded = await this.hasUploaded$.pipe(first()).toPromise();
-    if (hasUploaded) {
-      this.oldProofHash$
-        .pipe(
-          first(),
-          concatMap(oldProofHash =>
-            this.router.navigate(
-              ['details', { type: 'capture', hash: oldProofHash }],
-              { relativeTo: this.route }
+    return this.isCollecting$
+      .pipe(
+        first(),
+        switchMap(isCollecting =>
+          iif(
+            () => !isCollecting,
+            this.oldProofHash$.pipe(
+              first(),
+              concatMap(oldProofHash =>
+                this.router.navigate(
+                  ['details', { type: 'capture', hash: oldProofHash }],
+                  { relativeTo: this.route }
+                )
+              )
             )
-          ),
-          untilDestroyed(this)
-        )
-        .subscribe();
-    } else {
-      this.toastController
-        .create({
-          message: this.translocoService.translate(
-            'home.profileTab.captureNotUploadedYet'
-          ),
-          duration: 2000,
-          color: 'primary',
-        })
-        .then(toast => toast.present());
-    }
+          )
+        ),
+        untilDestroyed(this)
+      )
+      .subscribe();
   }
 }
```

Comment:
ChatGPT Plus code review with improvement suggestions:

1.  Add import for `switchMap` operator: It seems like you are using the `switchMap` operator in the code, but it has not been imported. Add the import statement for the `switchMap` operator at the beginning of the file:

```typescript
    import { switchMap } from 'rxjs/operators';
```

2.  Improve readability with parentheses: For better readability, consider adding parentheses around the arrow function in the `iif` operator:

```typescript
    () => !isCollecting
```

    Change to:

```typescript

    () => (!isCollecting)
```

3.  Remove unnecessary `return` statement: The `return` statement in the `onClick()` method is not necessary. You can remove it and directly use the pipeline instead:

```typescript
async onClick() {
   this.isCollecting$
```

4.  Add comments to describe the functionality: Adding comments throughout the code can help explain the purpose of specific sections, making it easier for others to understand. Consider adding comments to describe the functionality of the `onClick()` method and other important parts of the code.

Other than these suggestions, your code looks clean and well-structured.